completion_aggregator/tasks.py

```diff
@@ -12,28 +13,32 @@
 from xblock.completable import XBlockCompletionMode
 from xblock.core import XBlock
 
-from django.conf import settings
-
+from . import compat
 from .models import Aggregator
 
 OLD_DATETIME = pytz.utc.localize(datetime(1900, 1, 1, 0, 0, 0))
 
+log = logging.getLogger(__name__)
+
 
 CompletionStats = namedtuple('CompletionStats', ['earned', 'possible', 'last_modified'])  # pylint: disable=invalid-name
 
 
 @shared_task
-def update_aggregators(user, course_key, block_keys=frozenset()):  # pylint: disable=unused-argument
+def update_aggregators(user, course_key, block_keys=(), force=False):  # pylint: disable=unused-argument
     """
     Update aggregators for the specified course.
 
     Takes a collection of block_keys that have been updated, to enable
     future optimizations in how aggregators are recalculated.
     """
     from xmodule.modulestore.django import modulestore   # pylint: disable=import-error
+    block_keys = set(block_keys)
+
+    log.debug("Updating aggregators in %s for %s", course_key, user)
 
     updater = AggregationUpdater(user, course_key, modulestore())
-    updater.update(block_keys)
+    updater.update(block_keys, force)
```

Comment:
If you move the bulk of this method into a private method (e.g. `_update_aggregators`), then you can test its functionality in `tests.test_tasks` without having to go through the trouble of mocking celery.

Reply:
Good thinking.